kotlinx-coroutines-core/common/test/LimitedParallelismSharedTest.kt

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2016-2022 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.coroutines
+
+import kotlin.test.*
+
+class LimitedParallelismSharedTest : TestBase() {
+    @Test
+    fun testTaskFairness() = runTest {
+        val view = Dispatchers.Default.limitedParallelism(1)
+        val view2 = Dispatchers.Default.limitedParallelism(1)
+        val j1 = launch(view) {
+            while (true) {
+                yield()
+            }
+        }
+        val j2 = launch(view2) { j1.cancel() }
+        joinAll(j1, j2)
+    }
```

Comment:
Suggestion: move this test to the JS source-set, and move the `testTaskFairness` test from the JVM source-set to `concurrent`. This way, there will still be two versions of the test, but each one will be the most specific one for each target.

Reply:
Left this one as is to still have common coverage and added test with `newSingleThreadContext` to `concurrent` source set. Juggled with names a bit as well

Reply:
Ok, but I think having this test looks confusing. Maybe add a comment?

Reply:
Sure, done